src/daft-logical-plan/src/ops/window.rs

```diff
@@ -54,14 +55,23 @@ impl Window {
     ) -> Result<Self> {
         let input_schema = input.schema();
 
-        let mut fields = input_schema.fields.clone();
-
-        for (name, expr) in aliases.iter().zip(window_functions.iter()) {
-            let dtype = expr.to_field(&input_schema)?.dtype;
-            fields.insert(name.to_string(), Field::new(name.to_string(), dtype));
-        }
-
-        let schema = Arc::new(Schema::new(fields.values().cloned().collect())?);
+        let fields = input_schema
+            .fields()
+            .iter()
+            .cloned()
+            .map(Ok)
+            .chain(
+                aliases
+                    .iter()
+                    .zip(window_functions.iter())
+                    .map(|(name, expr)| {
+                        let dtype = expr.to_field(&input_schema)?.dtype;
+                        Ok(Field::new(name, dtype))
+                    }),
+            )
+            .collect::<DaftResult<_>>()?;
+
+        let schema = Arc::new(Schema::new(fields));
```

Comment:
This is another interesting recurring pattern I'm seeing, rather than having to wrap in oks, or chaining we can have a `schema.concat(&self, other: SchemaRef) -> SchemaRef` or similar signature. This way we can simply compute the window schema and concatenate to the input schema. This can also be re-used elsewhere as I've noticed similar logic in other places.

Reply:
We do currently have `Schema::union` and `Schema::non_distinct_union`. I do eventually want to deprecate these though since they rely on naming uniqueness. I think once we start building stuff with ordinals, I'll add a `Schema::concat`

Reply:
But if it's alright, I don't plan on making those changes in this PR in order to keep the changes small